standalone-tools/wss-scan.sh

```diff
@@ -0,0 +1,121 @@
+###### Information ############################################################################
+# Name:          wss-scan.sh
+# Maintainer:    ODFE Infra Team
+# Language:      Shell
+#
+# About:         This script is to scan the odfe distros for vulnerabilities and licenses
+#                It will scan the repositories and send the WhiteSource link to the mail 
+#                of the user. 
+#
+# Prerequisites: Need to set the Java Path
+#                Need to set the recepient mail in wss-scan.config
+#
+# Usage:         ./wss-scan.sh
+#
+# Starting Date: 10-13-2020
+# Modified Date: 10-15-2020
+###############################################################################################
+
+#Download the WhiteSource Agent 
+curl -LJO -sS https://github.com/whitesource/unified-agent-distribution/releases/latest/download/wss-unified-agent.jar > /dev/null
+
+echo "Download completed"
+
+#Scanning the config file for the user configurations
+source wss-scan.config
+gitRepos=${gitRepos//,/$'\n'}  # change the semicolons to white space
+
+basepath=$baseDirPath"/repos"
+
+if [[ ! -e $basepath ]]; then
+    mkdir -p $basepath
+fi
+ 
+
+echo "Cleaning up scan directories if already present"
+rm -rf $basepath/*
+
+
+#Cloning the desired Repos for scanning 
+for repo in $gitRepos
+do
+        echo "Cloning repo "$gitBasePath$repo
+    git clone "$gitBasePath$repo".git $basepath"/"$repo
+done
+
+cp /dev/null info.txt
+
+#Scanning the Repos using the WhiteSource Unified Agent
+for repo in $gitRepos
+do
+	if [ -d "$basepath"/"$repo" ]
+        then
+           echo "Scanning repo :"$gitBasePath$repo " Project:" $repo 
+	   repo_path=$basepath"/"$repo
+	   java -jar wss-unified-agent.jar -c wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product ODFE -project $repo | grep "Project name" | sed 's/^.\{,41\}//' >> info.txt 2>&1 &
+	else
+	   echo "Scanning failed for repo :"$gitBasePath$repo " Project:" $repo
+        fi
+done
+
+
+
+#Waiting for the scannings to complete
+while ps ax | grep -vw grep| grep -w "wss-unified-agent.jar" > /dev/null
```

Comment:
This can be solved by simply

```
ps ax | grep -w [w]ss-unified-agent.jar

Reply:
> Or just `wait`.

That also works, but for some reason I rarely use wait, probably because checking the ps directly seems more direct to me than not knowing if wait actually works or not.

Reply:
Included a condition to wait for sub processes and break after 70 mins.

Reply:
Are you still planning to use the `ps ax | grep -vw grep| grep -w "wss-unified-agent.jar"`
And you can just use `-q` to ignore output.
This is a quick cumbersome approach in your script. I dont approve this method here.

Reply:
Unless you *know* that `wait` is not going to work, just use it. It is the simplest solution here.

Reply:
I agree wait is the simplest solution, but I think the wait might fail in some scenarios. Just wanted to keep the logic in, to confirm that there is a break.